libs/ledger-live-common/src/families/bitcoin/wallet-btc/wallet.ts

```diff
@@ -269,35 +283,35 @@ class BitcoinLikeWallet {
       ];
     });
 
-    const lastOutputIndex = txInfo.outputs.length - 1;
+    const lastOutputIndex = orderedOutputs.length - 1;
 
     log("hw", `createPaymentTransaction`, {
       inputs,
       associatedKeysets,
       outputScriptHex,
-      ...(params.lockTime && { lockTime: params.lockTime }),
-      ...(params.sigHashType && { sigHashType: params.sigHashType }),
-      ...(params.segwit && { segwit: params.segwit }),
+      lockTime: params.lockTime,
+      sigHashType: params.sigHashType,
+      segwit: params.segwit,
       initialTimestamp,
-      ...(params.expiryHeight && { expiryHeight: params.expiryHeight }),
-      ...(txInfo.outputs[lastOutputIndex]?.isChange && {
-        changePath: `${fromAccount.params.path}/${fromAccount.params.index}'/${txInfo.changeAddress.account}/${txInfo.changeAddress.index}`,
-      }),
+      expiryHeight: params.expiryHeight,
+      changePath: orderedOutputs[lastOutputIndex]?.isChange
+        ? `${fromAccount.params.path}/${fromAccount.params.index}'/${txInfo.changeAddress.account}/${txInfo.changeAddress.index}`
+        : undefined,
       additionals: additionals || [],
     });
 
     const tx = await btc.createPaymentTransaction({
       inputs,
       associatedKeysets,
       outputScriptHex,
-      ...(params.lockTime && { lockTime: params.lockTime }),
-      ...(params.sigHashType && { sigHashType: params.sigHashType }),
-      ...(params.segwit && { segwit: params.segwit }),
+      lockTime: params.lockTime,
```

Comment:
in theory, there should not be impact for this simplification. 
Launch the bot and confirm that there is no breaking change.
Thanks a lot.

Reply:
I agree with being cautious with that change :warning:
That weird pattern was here for a reason, and it was because providing these params as `undefined` caused some errors, and it was required to _not_ provide them instead...

Reply:
From a quick look at the `createPaymentTransaction` function, its implem in _BtcNew.ts_ looks safe, but its implem in _BtcOld.ts_ doesn't. So I think it should stay like it is for now...

Reply:
Reverted and added a comment